systems/plants/@RigidBodyManipulator/doKinematics.m

```diff
@@ -10,64 +10,57 @@
 % the answer).
 %
 
-if model.use_new_kinsol
-  checkDirty(model);  
-  if nargin<6
+checkDirty(model);
+
+% method signature transition
+if nargin > 3
+  options_copy = options;
+end
+warn_signature_changed = false;
+if nargin == 5
+  warn_signature_changed = true;
+  options = struct();
+  options.compute_gradients = v;
```

Comment:
It depends on what you view as a gradient and whether it's about computing gradients of things inside doKinematics or the ability to compute gradients of things using the results of the doKinematics call.
I named the option compute_gradients in doKinematics because when it's set to true, doKinematics computes gradients of everything that would normally be put in kinsol (what I consider the base level).
The base level allows you to compute geometric Jacobians, which aren't really gradients of anything, and also forwardKin Jacobians, which are. But I also came up with an algorithm (currently not in use) that allows you to compute gradients of geometric Jacobians using only the information in a kinsol from a non-gradient doKinematics call. A non-gradient kinsol also allows you to compute a mass matrix, which can be considered a base level thing, or the Hessian of kinetic energy w.r.t. velocity. I know that's kind of a stretch, but you see my point that if you consider the ability to compute gradients of things outside of doKinematics, it's not 100% clear anymore.
Any suggestions for what the name should be if you still think it's confusing?